host/src/containers/GameSessionContainer.tsx

```diff
@@ -104,20 +112,66 @@ const GameSessionContainer = () => {
     }
   }, [isModalOpen]);
 
+  useEffect(() => { //headerGame timer, 1 second interval, update localstorage in case page resets
+    if (gameTimer && !gameTimerZero){
+      let refreshIntervalId = setInterval(() => {
+        if (headerGameCurrentTime > 0) {
+          setHeaderGameCurrentTime(headerGameCurrentTime - 1);
+          localStorage.setItem('currentGameTimeStor', headerGameCurrentTime-1);
+        }
+        else
+          setGameTimerZero(true);
+      }, 1000);
+      return () => clearInterval(refreshIntervalId);
+    }
+  }, [gameTimer, gameTimerZero, headerGameCurrentTime]);
+
+  useEffect(()=>{ //update gameSession currentTimer every 3 seconds with time from headerGame Timer 
+    if (gameTimer && !gameTimerZero){
+      let refreshIntervalId = setInterval(() => {
+          let newUpdates = {currentTimer: (localStorage.getItem('currentGameTimeStor')>= 0 ? localStorage.getItem('currentGameTimeStor') : 0)};
+          apiClient.updateGameSession({ id: gameSessionId, ...newUpdates });
+      }, 3000);
```

Comment:
We should test with different numbers here, especially with mobile devices and the internet access, there can be delays. 3s might just work or be too many updates

Reply:
100% agree, I think we'll need to build and test with mobile to find the acceptable frequency